### PR TITLE
[posix] enable expect tests on macOS

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: .github/workflows/posix.yml [hunk 1]
```diff
@@ -41,6 +41,33 @@ jobs:
...
+      CFLAGS: -DCLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER=1 -DOPENTHREAD_CONFIG_MLE_MAX_CHILDREN=15
+      CXXFLAGS: -DCLI_COAP_SECURE_USE_COAP_DEFAULT_HANDLER=1 -DOPENTHREAD_CONFIG_MLE_MAX_CHILDREN=15
+    steps:
+    - uses: actions/checkout@v2
+    - name: Bootstrap
+      run: |
+        brew install ninja
+    - name: Codecov
+      uses: codecov/codecov-action@v1
+    - name: Run RCP Mode
```
Comment (jwhui) [inline .github/workflows/posix.yml:56]: Both runs use RCP. Can we better differentiate?

File: .github/workflows/simulation.yml [hunk 2]
```diff
@@ -204,15 +204,15 @@ jobs:
     - uses: actions/checkout@v2
     - name: Bootstrap
       run: |
-        OT_OPTIONS=-DOT_READLINE=OFF sudo apt-get --no-install-recommends install -y expect ninja-build
+        sudo apt-get --no-install-recommends install -y expect ninja-build
     - name: Run
       run: |
-        OT_OPTIONS=-DOT_TIME_SYNC=ON VIRTUAL_TIME=0 ./script/test build expect
+        OT_OPTIONS=-DOT_TIME_SYNC=ON ./script/test build expect
```
Comment (jwhui) [inline .github/workflows/simulation.yml:210]: What determines whether or not we enable virtual time?

File: .github/workflows/simulation.yml [hunk 3]
```diff
@@ -204,15 +204,15 @@ jobs:
...
     - name: Codecov
       uses: codecov/codecov-action@v1
     - name: Run RCP Mode
       run: |
-        OT_OPTIONS=-DOT_READLINE=OFF VIRTUAL_TIME=0 NODE_MODE=rcp ./script/test clean build expect
+        OT_OPTIONS=-DOT_READLINE=OFF NODE_MODE=rcp ./script/test clean build expect
```
Comment (jwhui) [inline .github/workflows/simulation.yml:215]: What determines whether or not we enable readline?

**Comment by bukepo** [comment]
This PR depends on #5392

File: .github/workflows/simulation.yml (hunk 2 above)
Comment (bukepo) [reply .github/workflows/simulation.yml:210]: The script now automatically set VIRTUAL_TIME=0 when running expect tests.

File: .github/workflows/simulation.yml (hunk 3 above)
Comment (bukepo) [reply .github/workflows/simulation.yml:215]: With readline enabled, many unexpected tty control characters will occur which often disturbs parsing command results.

File: .github/workflows/simulation.yml (hunk 2 above)
Comment (jwhui) [reply .github/workflows/simulation.yml:212]: Makes sense. We should note this in the script docs:
https://github.com/openthread/openthread/blob/1ad22bac21f01597d86bf6f52aed6fd1a1b00d46/script/test#L270

**Comment by bukepo** [comment]
There are still some unstable issues on macOS. I will get back to this PR later.

**Review by jwhui — APPROVED** [review]

**Pull request merged**